sklearn/kernel_ridge/kernel_ridge.py

```diff
@@ -143,23 +123,41 @@ def fit(self, X, y=None, sample_weight=None):
         self : returns an instance of self.
         """
         # Convert data
-        X, y = check_X_y(X, y, accept_sparse=("csr", "csc"), multi_output=True,
-                         y_numeric=True)
         if sample_weight is not None and not isinstance(sample_weight, float):
             sample_weight = check_array(sample_weight, ensure_2d=False)
 
         K = self._get_kernel(X)
         alpha = np.atleast_1d(self.alpha)
 
-        ravel = False
-        if len(y.shape) == 1:
-            y = y.reshape(-1, 1)
-            ravel = True
-
         copy = self.kernel == "precomputed"
-        self.dual_coef_ = _solve_cholesky_kernel(K, y, alpha,
+
+        ravel = False
+        if self.regression is True:
+            X, y_ = check_X_y(X, y, accept_sparse=("csr", "csc"),
+                              multi_output=True, y_numeric=True)
+            if len(y_.shape) == 1:
+                y_ = y_.reshape(-1, 1)
+                ravel = True
+        else:  # Multilabel classification
+            X, y = check_X_y(X, y)
+            # Check alpha
+            if self.alpha <= 0:
+                raise ValueError('alpha must be positive')
+
+            # y is going to be encoded by zero arrays
+            # with 1 in a position assigned to a label
+            self.label_encoder_ = LabelBinarizer()
+            y_ = self.label_encoder_.fit_transform(y)
+            self.classes_ = self.label_encoder_.classes_
+
+            # This value is similar to number of neurons in hidden
+            # layer in neural version of Extreme Learning Machine
+            self.h_ = K.shape[1]
```

Comment:
Isn't `self.h_ == self.fit_X_.shape[0]`?? Why do we need to store it separately? Also, `h_` is a terrible name.

Reply:
Sorry, that comes from my previous pull request about ELM, because this value was the equivalent to the number of neurons from the hidden layer in neural network version of Extreme Learning Machine... but more and more it does not make sense to me. I'll remove it